Analytical_Engine/Query/Graph/Depth.cs

```diff
@@ -93,6 +93,12 @@ public static partial class Query
         [Output("depths", "A Dictionary of the depths of the entities in the Graph.")]
         public static Dictionary<Guid, int> Depth(this Graph graph, Guid startEntity)
         {
+            if(graph == null)
+            {
+                BH.Engine.Reflection.Compute.RecordError("Cannot query the depth of a null graph.");
+                return null;
```

Comment:
To follow a single pattern:
```suggestion
                return new Dictionary<Guid, int>();
```